cdap-docs/cdap-apps/source/hydrator/creating.rst

```diff
@@ -193,7 +193,275 @@ property replaces the ``schedule`` property of an ETL Batch Application.
+                "schema": "{
+                  \"type\":\"record\",
+                  \"name\":\"etlSchemaBody\",
+                  \"fields\":[
+                    {\"name\":\"userId\",\"type\":\"string\"}
+                  ]
+                }",
+                "name": "targetCustomers",
+                "schema.row.field": "userId"
+              }
+            }
+          }
+        ],
+        "transforms": [
+          {
+            "name": "spendingUsersScript",
+            "plugin": {
+              "name": "Script",
+              "properties": {
+                "script": "function transform(input, context) {
+                            if (input.purchasePrice > 1000) {
+                              return {'userId' : input.userId};
+                            }
+                          }",
+                "schema": "{
+                  \"type\":\"record\",
+                  \"name\":\"etlSchemaBody\",
+                  \"fields\":[
+                    {\"name\":\"userId\",\"type\":\"string\"}
+                  ]
+                }"
+              }
+            }
+          }
+        ],
+        "connections": [
+          {
+            "from": "purchaseStats",
+            "to": "replicaTable"
+          },
+          {
+            "from": "purchaseStats",
+            "to": "spendingUsersScript"
+          },
+          {
+            "from": "spendingUsersScript",
+            "to": "usersTable"
+          }
+        ],
+        "schedule": "* * * * *",
+        "engine": "mapreduce"
+      }
+    }
+
+**Note:**
+Pipeline connections can be configured to fork from a stage, with the output of the stage sent to
+two or more configured stages; in the above example, the same output record is sent to the other stages.
+Similarly, forked transform stages can merge together at a transform or a sink stage.
+The output schema's from these merging stages must match before joining. Note that
+the order of records sent from the forked stages to the joining stage will not be defined.
+
+In this next example, ``purchaseStream`` has purchase data with fields ``userid``, ``item``, ``count``, and ``price``.
+The stream events source stage ``purchaseStream`` forks, and records are sent to both of the
+transforms ``userRewards`` and ``itemRewards``.
+
+The ``userRewards`` transform script looks up valued customers in the table ``hvCustomers``,
+to check if ``userid`` is a valued customer and assigns higher rewards if they are.
+After calculating the rewards, this transform sends an output record in the format ``userid(string), rewards(double)``.
+
+The ``itemRewards`` transform script awards higher rewards for bulk purchases and sends output records in the 
+same format, ``userid(string), rewards(double)``.
+
+The rewards records are merged at the sink ``rewardsSink``; note that the incoming schema from the transforms
+``userRewards`` and ``itemRewards`` are the same, and that the order of received records will vary.
+
```

Comment:
also would be very useful to have a diagram here. Otherwise its just a wall of text that is hard to understand